### PR TITLE
WIP: Stop duplicating ApplicationPolicy#initialize in the subclasses

### PR DESCRIPTION
**Pull request by Ana06**
Use the parent method instead of override it with the same code. :bowtie:

### CODE REVIEW

**Comment by ChrisBr** [comment]
> it is due to this line:

Exactly, it is because sometimes user is required and sometimes not. User is required always in the Application policy therefore we need to overwrite it. Might make sense to do it the other way around, yes.

File: src/api/app/policies/project_policy.rb [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   def initialize(user, record)
-    raise Pundit::NotAuthorizedError, 'record does not exist' unless record
+    require_record(record)
     @user = user
     @record = record
   end
 
   def create?
-    return false unless @user
+    require_user(user)
```
Comment (coolo) [inline src/api/app/policies/project_policy.rb]: there is a difference between return false and raising an exception. This is not create! but create?

File: src/api/app/policies/project_policy.rb [hunk 2]
```diff
@@ -1,17 +1,17 @@
 class ProjectPolicy < ApplicationPolicy
   def initialize(user, record)
-    raise Pundit::NotAuthorizedError, 'record does not exist' unless record
+    require_record(record)
```
Comment (coolo) [inline src/api/app/policies/project_policy.rb]: I think I'd prefer a call to super with flags if *not* required.

**Comment by Ana06** [comment]
DO NOT MERGE: WIP

File: src/api/app/policies/project_policy.rb (hunk 1 above)
Comment (Ana06) [reply src/api/app/policies/project_policy.rb]: There is no difference. The gem will raise an exception if it return `false`. So it doesn't matter if you raise it first. Check: https://github.com/varvet/pundit/blob/master/lib/pundit.rb#L70

File: src/api/app/policies/project_policy.rb (hunk 2 above)
Comment (Ana06) [reply src/api/app/policies/project_policy.rb]: how? :thinking:

**Comment by Ana06** [comment]
It seems like I don't find time to finish this. I'll close for now.

**Pull request closed without merging**